standalone-metastore/metastore-common/src/gen/thrift/gen-cpp/hive_metastore_types.cpp

```diff
@@ -615,16 +615,18 @@ int _kTxnTypeValues[] = {
   TxnType::REPL_CREATED,
   TxnType::READ_ONLY,
   TxnType::COMPACTION,
-  TxnType::MATER_VIEW_REBUILD
+  TxnType::MATER_VIEW_REBUILD,
+  TxnType::SOFT_DELETE
```

Comment:
Will this cause any compatibility issues?

Reply:
Old versions of HMS would treat new values as null (default - null), in this case, we won't ever enter the if branch for SOFT_DELETE, meaning, we won't populate the highest write id required by cleaner and might clean up some deltas prematurely, see HIVE-24291.

Reply:
however, old HMS won't have support for lockless read, so we should be good